### PR TITLE
Fix Minimum Wage

### PR DESCRIPTION
**Pull request by mgrassotti**
### Description
**Trello**: https://trello.com/c/IonCW10l/660-investigate-minimum-wage-smart-answer
**Zendesk Ticket**: https://govuk.zendesk.com/agent/tickets/2231061

Calculate the basic_rate as the average between the normal rate and
the overtime_hourly_rate, instead of just returning the overtime_hourly_rate.

This is stated here: https://www.gov.uk/overtime-your-rights/overview.

This prevents the message "You are not getting the minimum wage" to appear, if
the basic rate is above the minimum rate.

[GOV.UK](https://www.gov.uk/am-i-getting-minimum-wage/y/current_payment/not_an_apprentice/40/30/37.0/4000.0/1.0/3.0/no)
[Preview](https://smart-answers-preview-pr-3230.herokuapp.com/am-i-getting-minimum-wage/y/current_payment/not_an_apprentice/40/30/37.0/4000.0/1.0/3.0/no)

### Notes for Factcheck
This introduces the calculation of the rate as the average between the normal rate and
the overtime_hourly_rate. Apart from avoiding the result "You are not getting the minimum wage" to be shown, it introduces a change in a lot of outcomes, due to this different calculation. They should be reviewed and approved too.
Some examples of paths with changes: 
Before: https://www.gov.uk/am-i-getting-minimum-wage/y/current_payment/apprentice_over_19_first_year/1/16.0/300.0/8.0/10.0/no
After: https://smart-answers-preview-pr-3230.herokuapp.com/am-i-getting-minimum-wage/y/current_payment/apprentice_over_19_first_year/1/16.0/300.0/8.0/10.0/no

Before: https://www.gov.uk/am-i-getting-minimum-wage/y/current_payment/apprentice_over_19_first_year/1/16.0/300.0/8.0/10.0/yes_charged/5.0/5
After: https://smart-answers-preview-pr-3230.herokuapp.com/am-i-getting-minimum-wage/y/current_payment/apprentice_over_19_first_year/1/16.0/300.0/8.0/10.0/yes_charged/5.0/5)

Before: https://www.gov.uk/am-i-getting-minimum-wage/y/current_payment/apprentice_over_19_first_year/1/16.0/300.0/8.0/10.0/yes_charged/50.0/5
After: https://smart-answers-preview-pr-3230.herokuapp.com/am-i-getting-minimum-wage/y/current_payment/apprentice_over_19_first_year/1/16.0/300.0/8.0/10.0/yes_charged/50.0/5

### Before
screen shot 2017-10-12 at 11 39 59

### After
screen shot 2017-10-12 at 11 43 00

### CODE REVIEW

File: lib/smart_answer/calculators/minimum_wage_calculator.rb [hunk 1]
```diff
@@ -54,7 +54,7 @@ def valid_age_for_living_wage?(age)
     def basic_rate
       rate = @basic_pay / @basic_hours
       if overtime_hours > 0 && overtime_hourly_rate > 0 && rate > overtime_hourly_rate
-        overtime_hourly_rate
+        (rate + overtime_hourly_rate) / 2
```
Comment (chao-xian) [inline lib/smart_answer/calculators/minimum_wage_calculator.rb]: I don't understand what the `2` represents. Please can we avoid introducing more magic numbers.

Comment (mgrassotti) [reply lib/smart_answer/calculators/minimum_wage_calculator.rb]: @chao-xian it is the average between the two, this is what I understood from https://govuk.zendesk.com/agent/tickets/2231061 and https://www.gov.uk/overtime-your-rights/overview. I added a comment to make it more clear in the code.

Comment (chao-xian) [reply lib/smart_answer/calculators/minimum_wage_calculator.rb]: Could this be expressed by assigning to a variable named appropriately? Something like `average_rate` and then you wouldn't need to add the comment. How do you feel about that?

Comment (mgrassotti) [reply lib/smart_answer/calculators/minimum_wage_calculator.rb]: @chao-xian 👍

Comment (mgrassotti) [reply lib/smart_answer/calculators/minimum_wage_calculator.rb]: @chao-xian mmh, Linter complains about that, I simplified the comment

**Comment by chao-xian** [comment]
@mgrassotti checksums need updating

**Comment by mgrassotti** [comment]
@chao-xian thanks, it's done

**Comment by mgrassotti** [comment]
@chao-xian is that ok for you now? thanks!

**Pull request merged**